types/lodash/index.d.ts

```diff
@@ -12269,7 +12269,7 @@ declare namespace _ {
          * @param value The value to inspect.
          * @return Returns true if value is empty, else false.
          */
-        isEmpty(value?: any): boolean;
+        isEmpty(value?: any): value is null | undefined;
```

Comment:
I'm with @andy-ms on this one - there is a much simpler case that will be problematic:

```
function ensureNotEmpty(arr: string[]) {
    if (_.isEmpty(arr)) {
        arr.push("foo"); // the compiler fails here because arr has type never
    }
}
```

I can see how it would be nice to get a proper type guard for this, but if the type guard handles any cases incorrectly, I think it would be better to just return `boolean` and use `!` when necessary.